### PR TITLE
[WIP] Comms error handling bug fix

### PR DESCRIPTION
**Pull request by pazaan**
Sets retries back to zero after issuing a "ResendPacketsCommand"
so that we don't spam the output command queue on the USB stick
with requests.
Doing so causes the stick to get buffer issues, and also means that we
double up on getting packets that we already had requested.

Also throw an InvalidStateError if we get packets we already received,
to make it really obvious that something is still not quite right.

### CODE REVIEW

**Comment by pazaan** [comment]
[When the 670G sensor has blocked communication with the pump during upload, the upload fails.](https://trello.com/c/XLio0cA6/383-when-the-670g-sensor-has-blocked-communication-with-the-pump-during-upload-the-upload-fails)

**Comment by pazaan** [comment]
@gniezen This is a definite improvement, but I'll leave this branch open in case I make further improvements.

File: lib/drivers/medtronic600/medtronic600Driver.js [hunk 1]
```diff
@@ -988,6 +991,14 @@ class TransmitPacketRequest extends NGPRequest {
           this.ngpRetries += 1;
         }
         debug(`Multipacket retry: ${this.ngpRetries}`);
```
Comment (gniezen) [inline lib/drivers/medtronic600/medtronic600Driver.js]: I guess this line can now be removed, as it looks like a duplicate of line 995?

Comment (pazaan) [reply lib/drivers/medtronic600/medtronic600Driver.js]: Removed the duplicate information from line 995.

**Pull request merged**